javascripts/discourse/components/search-banner.js

```diff
@@ -1,16 +1,21 @@
-import Component from "@ember/component";
+import { action, computed } from "@ember/object";
+import { and } from "@ember/object/computed";
 import { inject as service } from "@ember/service";
+import Component from "@glimmer/component";
 import { defaultHomepage } from "discourse/lib/utilities";
-import { and } from "@ember/object/computed";
-import discourseComputed, { observes } from "discourse-common/utils/decorators";
 
-export default Component.extend({
-  router: service(),
-  tagName: "",
+export default class SearchBanner extends Component {
+  @service router;
+  @service siteSettings;
+  @service currentUser;
 
-  @discourseComputed("router.currentRouteName")
-  displayForRoute(currentRouteName) {
+  @and("displayForUser", "displayForRoute") shouldDisplay;
+
+  @computed("router.currentRouteName")
+  get displayForRoute() {
```

Comment:
Using the legacy computed property pattern in a glimmer component is a bit of a code smell. IMO we should aim for native getters and rely on autotracking. In this case, I think you can just drop the `@computed` and it'll work fine 👌

Reply:
I was hoping the same thing, but for some reason it did not work as expected. :(

Reply:
Maybe because we're still using the legacy computed `@and`, which only works for computed properties? If we turn that into

```js
get shouldDisplay(){
  return this.displayForUser && this.displayForRoute;
}
```
then I think it should all work ok 🤞

Reply:
That worked well, thank you!